build-system/babel-plugins/babel-plugin-transform-fix-leading-comments/index.js

```diff
@@ -0,0 +1,40 @@
+/**
+ * Copyright 2020 The AMP HTML Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS-IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+// Reassigns the trailing comments of a statement to be leading comment of its
+// next sibling. This is because JSDoc comments (which should be on the next
+// statement) get erroneously assigned as trailing comments of this statement.
+module.exports = function() {
+  return {
+    visitor: {
+      Statement(path) {
+        const {node} = path;
+        const {trailingComments} = node;
+        if (!trailingComments) {
+          return;
+        }
+
+        const next = path.getNextSibling();
+        if (!next) {
+          return;
+        }
+
+        node.trailingComments = null;
+        next.addComments('leading', trailingComments);
```

Comment:
why do we want all trailing comments to become leading comments of the next node?

Reply:
So that inserted nodes don't get injected between a JSDoc comment (which would have been a trailing comment of the previous node) and the function it describes.